tests/js/firstDiscoveryEditorTests.js

```diff
@@ -34,20 +34,63 @@ https://github.com/gpii/universal/LICENSE.txt
         return builder.options.assembledPrefsEditorGrade;
     };
 
-    jqUnit.asyncTest("Initialization", function () {
+    gpii.tests.verifyStates = function (that, currentPanelNum, backVisible, nextVisible, panelsVisibility) {
+        var prefsEditorContainer = that.locate("prefsEditor");
+        var backButton = that.navButtons.locate("back");
+        var nextButton = that.navButtons.locate("next");
+
+        jqUnit.assertEquals("The model value for \"currentPanelNum\" has been set to " + currentPanelNum, currentPanelNum, that.model.currentPanelNum);
+        fluid.each(panelsVisibility, function (panelSelectors, state) {
+            fluid.each(panelSelectors, function (selector) {
+                jqUnit[state]("The visibility of the panel " + selector + " is " + state, prefsEditorContainer.find(selector));
+            })
+        });
+
+        jqUnit[backVisible ? "isVisible" : "notVisible"]("The visibility of the back button is " + backVisible, backButton);
+        jqUnit[nextVisible ? "isVisible" : "notVisible"]("The visibility of the next button is " + nextVisible, nextButton);
+    };
+
+    jqUnit.asyncTest("The first discovery tool editor", function () {
         gpii.tests.firstDiscovery("#gpiic-tool", {
```

Comment:
The current best practice is to avoid long configuration within a test. This should instead be moved to a new grade definition that you invoke within the asyncTest.